llvm/lib/CodeGen/MachineLICM.cpp

```diff
@@ -369,6 +377,36 @@ bool MachineLICMBase::runOnMachineFunction(MachineFunction &MF) {
   AA = &getAnalysis<AAResultsWrapperPass>().getAAResults();
 
   SmallVector<MachineLoop *, 8> Worklist(MLI->begin(), MLI->end());
+
+  // Initialize `AllowedToHoistLoads' if needed.
+  if (HoistConstLoads) {
+    auto TmpWorklist = Worklist;
+    // Initialize all loops with true values
+    while (!TmpWorklist.empty()) {
+      auto *L = TmpWorklist.pop_back_val();
+      AllowedToHoistLoads[L] = true;
+      TmpWorklist.insert(TmpWorklist.end(), L->getSubLoops().begin(),
```

Comment:
Oh, does MLI not already include subloops? I was hoping you could get away with just:

```
for (auto *L : Worklist) {
  AllowedToHoistLoads[L] = true;
}
```

It's quite likely that was a bogus assumption. :)

Reply:
Yeah, at least it is the same behaviour as with IR Loops